### ISSUE TITLE
Make map markers focusable

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#23046]
## Describe the defect
Currently map markers are not part of tab order at all. They can't be reached by keyboard navigation. Markers can be clicked, so they should be focusable as well. Additionally, markers currently don't have accessible names. 

The marker is currently marked up using a span
`<span style="cursor: pointer; transform: translate(-50%, -50%) translate(316px, 596px) rotateX(0deg) rotateZ(0deg);" class="i-pin-card-map pin-1 mapboxgl-marker mapboxgl-marker-anchor-center">1</span>` 

## Recommendation
Changing the span to an interactive element will put the markers into the default tab order. Additionally, interactive elements take their accessible names from their content, so the markers would now have an accessibile name, '2'.

- Change the span to a button.
- This should cover make the marker tabbable by default and the button will automatically use the text content (the number) as the button name.
- Ensure distinct default, hover, focus, active states are implemented.

## Current behavior
- Desktop tab behavior
  - Tab order, after hitting submit runs through each link in each teaser, then to the map overall, then it looks like it focuses the center of the map, then up to map zoom controls, then to the series of links at the bottom of the map frame.
  - Then it goes back to the center of the map, zoom controls, and map links again before you tab out of the
  - Markers are never focused.
- Mobile tab behavior
  - Tab order, after hitting submit runs through each link in each teaser, then to the map overall, then it looks like it focuses the center of the map, then up to map zoom controls, then to the mapbox at the bottom of the map frame before tabbing out of map.
  - Same order as desktop, but did not repeat tab stops for map controls/links.

## Notes
- There’s no need to have the repeat map control tab stops in desktop. (Bonus fix? If it's a small lift.)

## AC / Expected behavior
- [ ] When navigating with a keyboard, you can tab to the map markers.
- [ ] When using a screenreader and interacting with a marker, it reads the name (ie '2') 

## Screenshots

<details>
<summary>
Desktop tab stop order:
</summary>

<img width="1061" height="941" alt="Image" src="https://github.com/user-attachments/assets/e3dc4894-e520-4e6c-bc6b-016eb3cb35ee" />

</details>
<details>
<summary>
Mobile tab stop order:
</summary>

<img width="443" height="872" alt="Image" src="https://github.com/user-attachments/assets/ba95bbfe-1a05-4b05-9e05-212e5fe93d5b" />

</details>

## Definition of done
- [ ] Code completed; PR created
- [ ] Review instance created/available; internal team notified to start reviews
- [ ] Automated tests passed 
- [ ] Code review (meets established quality and security standards)
- [ ] PR approved
- [ ] Design review (meets design & UX requirements)
- [ ] A11y review (meets accessibility standards)
- [ ] Functional/feature review (meets acceptance criteria)
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by qjac** [comment]
@chriskim2311 @r-bartlett-gsa the markers are tabbable! In Chrome and Edge on CAG, I can tab through and see the focus outlines. The focus outlines are difficult to see on the map background. Might be good to ticket some design time to think through alternate/additional focus. 

Screenshot of focus outline on markers:
<img width="227" height="246" alt="Image" src="https://github.com/user-attachments/assets/58422a6d-fc35-40f0-9a44-c4b40bed98b2" />

Now that the markers are buttons, they are named by their content (ie '9'). So that's a big win as well!
